src/plugins/data/common/index_patterns/index_patterns/ensure_default_index_pattern.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { contains } from 'lodash';
+import { CoreStart } from 'kibana/public';
+import { IndexPatternsContract } from './index_patterns';
+
+export type EnsureDefaultIndexPattern = () => Promise<unknown> | undefined;
+
+export const createEnsureDefaultIndexPattern = (
+  core: CoreStart,
```

Comment:
lets not pass in whole core but just uiSettingsClient

Reply:
is there a reason this is wrapped in a function ? why not just export ensureDefaultIndexPattern ?

Reply:
So we can pass in dependencies without needing to pass them through the code that calls this function. Aside from less dependency passing (which is nice) this allows client specific code to be called from common code.

Reply:
I think I'd prefer to leave further refactoring of this code to future PRs. I'm drawing a rather arbitrary line since I need to do things backwards - provide the service and then clean up the code.